src/header/map.rs

```diff
@@ -270,6 +273,13 @@ struct Links {
     tail: usize,
 }
 
+/// Access to the `links` value in a slice of buckets.
+///
+/// It's important that no other field is accessed, since it may have been
+/// freed in a `Drain` iterator.
+#[derive(Debug)]
+struct RawLinks<T>(*mut [Bucket<T>]);
```

Comment:
Similarly, it looks like the pointer is never null (and is therefore never null checked); consider 
```suggestion
struct RawLinks<T>(ptr::NonNull<[Bucket<T>]>);
```
to encode that in the type system?